src/core/unit.c

```diff
@@ -4000,6 +4000,48 @@ static int kill_or_sigqueue(PidRef* pidref, int signo, int code, int value) {
         }
 }
 
+static int unit_kill_one(
+                Unit *u,
+                PidRef *pidref,
+                int signo,
+                int code,
+                int value,
+                sd_bus_error *error,
+                const char *type) {
+
+        int r;
+
+        assert(u);
+        assert(type);
+
+        if (!pidref_is_set(pidref))
+                return 0;
+
+        _cleanup_free_ char *comm = NULL;
+        (void) pidref_get_comm(pidref, &comm);
+
+        r = kill_or_sigqueue(pidref, signo, code, value);
+        if (r == -ESRCH)
+                return 0;
+        if (r < 0) {
+                /* Report this failure both to the logs and to the client */
+                if (!sd_bus_error_is_set(error))
```

Comment:
hmm, this is a bit weird. so far we never looked into this, we only wrote into this on error.

i'd avoid this, and instead just pass NULL for error as argument to unit_kill_one() unless ret == 0. I think that's more elegant and keeps things as they are in regard to reading the param...

(btw, I think we should adopt a new naming scheme: params that are initialized strictly on error, and never on access we should start prefixing with `reterr_`, to match the existing nomenclature of `ret_` for those which are initialized on success, and never on failure. I think that would make things nicely symmetric and easier to grok. thee sd_bus_error parameters should then be renamed to reterr_error or so... just an idea)

Reply:
So, `reterr_error` here? I stick to previous rule, i.e. `ret_error` here. I'd like to rename them in a later PR consistently.